iocore/cache/CacheRead.cc

```diff
@@ -717,8 +717,28 @@ CacheVC::openReadMain(int /* event ATS_UNUSED */, Event * /* e ATS_UNUSED */)
     bytes     = doc->len - doc_pos;
     if (is_debug_tag_set("cache_seek")) {
       char target_key_str[CRYPTO_HEX_SIZE];
-      key.toHexStr(target_key_str);
-      Debug("cache_seek", "Read # %d @ %" PRId64 "/%d for %" PRId64, fragment, doc_pos, doc->len, bytes);
+      Debug("cache_seek", "Read # %d @ %" PRId64 "/%d for %" PRId64 " %s", fragment, doc_pos, doc->len, bytes,
+            key.toHexStr(target_key_str));
+    }
+
+    if (bytes < 0) { // check for sporadic bug with oversized fragments
```

Comment:
Need a serious comment block here, to explain what's going on. It will not be obvious how this can trigger.

Reply:
I've had serious problems recreating the bug but we do have evidence of this showing up in production with HTTP objects.  Full fetch of these assets work but range requesting past ~1MB results in corrupt data.  We only found this by trying to migrate a delivery service from background_fetch over to the slice plugin.

Reply:
Put that in. Something like
"This shouldn't happen for HTTP assets but it does occasionally in production. This is a temporary fix to clean up broken objects until the root cause can be found. It must be the case that either the fragment offsets are incorrect or a fragment table isn't being created when it should be."